packages/cli/test/integration/variables.test.ts

```diff
@@ -0,0 +1,293 @@
+import * as EnvironmentHelpers from '@/environments/enviromentHelpers';
+
+import type { AuthAgent } from './shared/types';
+
+// mock that credentialsSharing is not enabled
+let app: Application;
+let ownerUser: User;
+let memberUser: User;
+let authAgent: AuthAgent;
+
+beforeAll(async () => {
+	app = await utils.initTestServer({ endpointGroups: ['variables'] });
+
+	utils.initConfigFile();
+
+	ownerUser = await testDb.createOwner();
+	memberUser = await testDb.createUser();
+
+	authAgent = utils.createAuthAgent(app);
+});
+
+beforeEach(async () => {
+	await testDb.truncate(['Variables']);
+	config.set('enterprise.features.variables', true);
+});
+
+afterAll(async () => {
+	await testDb.terminate();
+});
+
+// ----------------------------------------
+// GET /variables - fetch all variables
+// ----------------------------------------
+
+test('GET /variables should return all variables for an owner', async () => {
+	await Promise.all([
+		testDb.createVariable('test1', 'value1'),
+		testDb.createVariable('test2', 'value2'),
+	]);
+
+	const response = await authAgent(ownerUser).get('/variables');
+	expect(response.statusCode).toBe(200);
+	expect(response.body.data.length).toBe(2);
+});
+
+test('GET /variables should return all variables for a member', async () => {
+	await Promise.all([
+		testDb.createVariable('test1', 'value1'),
+		testDb.createVariable('test2', 'value2'),
+	]);
+
+	const response = await authAgent(memberUser).get('/variables');
+	expect(response.statusCode).toBe(200);
+	expect(response.body.data.length).toBe(2);
+});
+
+// ----------------------------------------
+// GET /variables/:id - get a single variable
+// ----------------------------------------
+
+test('GET /variables/:id should return a single variable for an owner', async () => {
+	const [var1, var2] = await Promise.all([
+		testDb.createVariable('test1', 'value1'),
+		testDb.createVariable('test2', 'value2'),
+	]);
+
+	const response1 = await authAgent(ownerUser).get(`/variables/${var1.id}`);
+	expect(response1.statusCode).toBe(200);
+	expect(response1.body.data.key).toBe('test1');
+
+	const response2 = await authAgent(ownerUser).get(`/variables/${var2.id}`);
+	expect(response2.statusCode).toBe(200);
+	expect(response2.body.data.key).toBe('test2');
+});
+
+test('GET /variables/:id should return a single variable for a member', async () => {
+	const [var1, var2] = await Promise.all([
+		testDb.createVariable('test1', 'value1'),
+		testDb.createVariable('test2', 'value2'),
+	]);
+
+	const response1 = await authAgent(memberUser).get(`/variables/${var1.id}`);
+	expect(response1.statusCode).toBe(200);
+	expect(response1.body.data.key).toBe('test1');
+
+	const response2 = await authAgent(memberUser).get(`/variables/${var2.id}`);
+	expect(response2.statusCode).toBe(200);
+	expect(response2.body.data.key).toBe('test2');
+});
+
+// ----------------------------------------
+// POST /variables - create a new variable
+// ----------------------------------------
```

Comment:
One missing test is if the user is over the quota for the number of variables. Although we won't be strictly enforcing, it's important to be tested.